### PR TITLE
Add ParseFMPURI for parsing URIs with schemes 'fmprpc' or 'fmprpc+tls'

### PR DESCRIPTION
**Pull request by jacobhaven**
Copied with minor modifications from
https://github.com/keybase/client/blob/master/go/service/rpc.go
https://github.com/keybase/client/blob/master/go/service/rpc_test.go

r? @patrickxb @maxtaco @jzila @akalin-keybase

### CODE REVIEW

**Comment by maxtaco** [comment]
:+1:  but @jzila maybe take a look too?

**Comment by jzila** [comment]
LGTM

File: fmp_uri.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+)
+
+type fmpURIScheme string
```
Comment (jzila) [inline fmp_uri.go]: This should be exported, since there are fields of this type that are exported.

File: fmp_uri.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+type FMPURI struct {
+	Scheme   fmpURIScheme
+	HostPort string
+	Host     string
+}
+
+var errInvalidFMPScheme = errors.New("invalid framed msgpack rpc scheme")
+var errNoHost = errors.New("missing host in framed msgpack rpc URI")
+
+func ParseFMPURI(s string) (*FMPURI, error) {
```
Comment (jzila) [inline fmp_uri.go]: Please add some godoc comments so we can slowly move towards `govet` passing on this repo.

**Pull request merged**